pkg/tsdb/cloudwatch/models/cloudwatch_query.go

```diff
@@ -264,6 +266,8 @@ func ParseMetricDataQueries(dataQueries []backend.DataQuery, startTime time.Time
 			return nil, &QueryError{Err: err, RefID: refId}
 		}
 
+		cwQuery.applyMacros(startTime, endTime)
+
 		cwQuery.migrateLegacyQuery(mdq, dynamicLabelsEnabled)
```

Comment:
Intuitively feels wrong that migrations are applied after `validateAndSetDefaults` and `applyMacros`, but that's something to look into another time.